utils/src/main/scala/za/co/absa/hermes/utils/HelperFunctions.scala

```diff
@@ -38,12 +38,21 @@ object HelperFunctions {
     * especially on a vary big dataframes.
     *
     * @param df A dataframe
+   *  @param dfsForArraysLength DataFrames with the same schema as `df` that are used to
+   *                             flatten arrays with maximum length among `df` and `dfsForArraysLengths`.
     * @return A new dataframe with flat schema.
     */
-  def flattenSchema(df: DataFrame): List[Column] = {
+  def flattenSchema(df: DataFrame, dfsForArraysLength: DataFrame*): List[Column] = {
     val fields = new mutable.ListBuffer[Column]()
     val stringFields = new mutable.ListBuffer[String]()
 
+    def calculateMaxArrayLength(path: String, structField: Option[StructField] = None): Int = {
+      val fullPath = path + structField.map(_.name).getOrElse("")
+      val maxLengths = (Seq(df) ++ Seq(dfsForArraysLength: _*))
+        .map(_.agg(max(expr(s"size($fullPath)"))).collect()(0).getInt(0))
```

Comment:
**Important:*** Realized during night after posting the previous review, that this will throw an exception, if any of the DataFrames is empty.
Probably using `.headOption.getOrElse(0)` is the safe solution.

Reply:
Yeah that's true. Done